corehq/apps/users/static/users/js/mobile_workers.js

```diff
@@ -321,6 +320,40 @@ hqDefine("users/js/mobile_workers",[
             return self.STATUS.SUCCESS;
         });
 
+        self.passwordSatisfyLength = ko.computed(function () {
+            if (!self.stagedUser()) {
+                return true;
+            }
+
+            if (self.stagedUser().force_account_confirmation()) {
+                return true;
+            }
+
+            if (self.stagedUser().force_account_confirmation_by_sms()) {
+                return true;
+            }
+
+            if (!self.useStrongPasswords()) {
+                // No validation
+                return true;
+            }
```

Comment:
I think we may want to think about changing the structure of this, as we've now duplicated logic between `passwordStatus` and `passwordSatisfyLength`, as well as made `passwordSatisfyLength` less intuitive. For example, if the user had a password that was too short, but it had the `force_account_confirmation_by_sms` flag set, would it make sense that a function called `passwordSatisfyLength` would return `true`?  I think what we're trying to say here is, 'Does the password need to be validated? If so, check if it meets the length and complexity requirements. Otherwise, ignore it' The validation check could likely live outside of this.

Reply:
This redundancy also puzzled me when I wrote this. But I'm still new to knockout and don't have a better solution. Can you give me a more specific suggestion?